### DISCUSSION TITLE
Time of Preconditioner's tasks

### DISCUSSION THREAD

**Discussion by jinca** [idaholab/moose#20757] [Q&A Modules: General] [answered]
Hello dear MOOSE group,

It is very clear for me to identify the memory spent by a preconditioner in an input file.
It is a chart where it is indicated the "Reports information only for process 0."

May I ask about the time spent for the preconditioner?
In the chart there is a part about Preconditioner Setup + Applying the PC
Is that the sum of those times, the total time spent for all the operation of the preconditioner?
e.g.
PCSetUp                3 1.0 1.2164e+00 1.0 0.00e+00 0.0 0.0e+00 0.0e+00 1.2e+01  0  0  0  0  1   0  0  0  0  1     0
PCApply              461 1.0 2.0880e+01 1.0 0.00e+00 0.0 0.0e+00 0.0e+00 0.0e+00  5  0  0  0  0   5  0  0  0  0     0

The time spent here is 1.2164 + 2.0880 seconds?
In the report of the ouput is also shown the total time of the execution:
The Main Stage: 4.0190e+02 :: 401.9 seconds.

Thanks in advance for the answer,
Julita

**Comment by GiudGiud** [comment]
Hello

This is more of petsc profiling than MOOSE. 
Please see the petsc docs https://petsc.org/release/docs/manualpages/Profiling/index.html

@fdkong if you have done this before could you please share some thoughts or resources

Guillaume

**Comment by fdkong** [comment, marked as answer]
> The time spent here is 1.2164 + 2.0880 seconds?

Correct. This is the total time on the preconditioner.  

Here is more information on how to interpret the result of "-log_view" https://petsc.org/release/docs/manual/profiling/

If you still have anything unclear after going through the above link, please feel free to ask .
